core/src/test/java/io/ably/lib/test/realtime/ConnectionManagerTest.java

```diff
@@ -729,8 +731,19 @@ public void onChannelStateChanged(ChannelStateChange stateChange) {
             /* Wait for both channels to reattach and verify state histories match the expected ones */
             attachedChannelWaiter.waitFor(ChannelState.attached);
             suspendedChannelWaiter.waitFor(ChannelState.attached);
-            assertEquals("Attached channel histories do not match", attachedChannelHistory, expectedAttachedChannelHistory);
-            assertEquals("Suspended channel histories do not match", suspendedChannelHistory, expectedSuspendedChannelHistory);
+            //wait for callbacks and lists to populate
+            try {
+                Thread.sleep(1000);
+            } catch (InterruptedException e) {
+            }
+            assertEquals("Attached channel histories do not match", expectedAttachedChannelHistory, attachedChannelHistory);
+            assertEquals("Suspended channel histories do not match", expectedSuspendedChannelHistory, suspendedChannelHistory);
+        } catch (AblyException e) {
+            e.printStackTrace();
+            fail("channels_are_reattached_after_reconnecting_when_statettl_plus_idleinterval_has_passed: Unexpected exception");
```

Comment:
Can you fail the test by passing the error message inside it ?
There must be some assertion that takes accepts exception as a parameter

Reply:
The thing is we should have `error` messages visible on the local/CI once test fails, so it's easy to debug failing test.

Reply:
I think this should do it https://github.com/ably/ably-java/pull/842/commits/d7ed41a600e1c15ce435c7ceb947c2aa4efaf7f0